text/0000-cargo-test-coverage.md

```diff
@@ -0,0 +1,366 @@
+graph. Since this is a Cargo specific feature, the Rust compiler will not need any updates.
+
+## Alternatives
+
+### Alternative 1: leave the existing feature
+
+Supporting this alternative would mean that no changes would be necessary to either Cargo
+or Rust. Getting instrumentation-based code coverge is still supported and would continue
+to work as it does today.
+
+The drawback for this option is that it would require setting the flag for all crates in
+the dependency graph, including upstream dependencies. This would also instrument all
+binaries and report on coverage for functions and branches that are not defined by the
+current crate with the potential of skewing coverage results.
+
+### Alternative 2: use a new manifest key instead of a CLI option
+
+Supporting this alternative would mean that changes would need to be made to the existing
+manifest format to possibly include a new section and/or key. A new `coverage` key could
+be added to the target section, `coverage = true`. This still has the added benefit of not
+requiring any changes to the Rust compiler itself and the feature could be scoped to Cargo only.
+
+The drawback for this option is that it could potentially add clutter to the `Cargo.toml`
+files. Adding this new section to every crate that a Rust developer wants to have instrumented
+would only add to all the data that is already contained within a toml file.
+
+### Alternative 3: use a RUSTC_WRAPPER program to selectively choose which crates to profile
+
+Supporting this alternative would mean that there wouldn't need to be any changes to Cargo
+at all.
+
+This would require creating a RUSTC_WRAPPER program specifically for selecting which crates to profile.
+This means more boiler plate code for each Rust developer that simply wants to profile their own crate.
+I believe the feature this RFC proposes would both be a cleaner solution long term and more in line
+with the Cargo workflow of potentially reading these kinds of behaviors from the `Cargo.toml` manifest
+file.
+
+### Alternative 4: use existing custom subcommands to run code coverage analysis
+
+Supporting this alternative would mean that there wouldn't need to be any changes to Cargo
+at all.
+
+There are multiple custom subcommands that exist today to achieve instrumenting Rust libraries
+analyzing code coverage metrics. A couple of examples would be cargo-llvm-cov and cargo-tarpaulin.
+Both of these custom subcommands are available via crates.io and support running tests with code
+coverage enabled and creating a coverage report to be viewed by a Rust developer.
+
+Cargo-llvm-cov currently leverages the `cargo-config` subcommand which is still unstable. To do so,
+cargo-llvm-cov sets the `RUSTC_BOOTSTRAP` environment variable to allow its usage from a stable
+toolchain. This is not a recommended approach especially for Rust developers that want to use
+such tools in production code.
```

Comment:
cargo-llvm-cov is now using [cargo-config2](https://github.com/taiki-e/cargo-config2) crate and is no longer depending on the unstable cargo-config subcommand.

> This is not a recommended approach especially for Rust developers that want to use
such tools in production code.

Btw, if you think this is a problem, you may want to open the issue to rust-analyzer that does similar things to the previous cargo-llvm-cov.